### ISSUE TITLE
Cannot boot with microcode_ctl 2:2.1-30.qubes1.fc25

### ISSUE THREAD

**Issue by p1gp1g** [QubesOS/qubes-issues#5897]
**Qubes OS version**
Qubes release 4.0 (R4.0)

**Affected component(s) or functionality**
Kernel, microcode

**Brief summary**
Can't boot with the microcode_ctl 2:2.1-30.qubes1.fc25.

**To Reproduce**
Hardware specific. CPU : Intel(R) Core(TM) i5-8265U

**Expected behavior**
Boot

**Actual behavior**
Freeze

**Additional context**
It freezes right after this part (from the boot with the working kernel) : 
```
(XEN) I/O virtualisation enabled
(XEN)  - Dom0 mode: Relaxed
(XEN) Interrupt remapping enabled
(XEN) nr_sockets: 1
(XEN) Enabled directed EOI with ioapic_ack_old on!
(XEN) ENABLING IO-APIC IRQs
(XEN)  -> Using old ACK method
(XEN) ..TIMER: vector=0xF0 apic1=0 pin1=2 apic2=-1 pin2=-1
(XEN) TSC deadline timer enabled
(XEN) Allocated console ring of 64 KiB.
(XEN) mwait-idle: MWAIT substates: 0x11142120
(XEN) mwait-idle: v0.4.1 model 0x8e
(XEN) mwait-idle: lapic_timer_reliable_states 0xffffffff
(XEN) mwait-idle: max C-state count of 8 reached
(XEN) VMX: Supported advanced features:
(XEN)  - APIC MMIO access virtualisation
(XEN)  - APIC TPR shadow
(XEN)  - Extended Page Tables (EPT)
(XEN)  - Virtual-Processor Identifiers (VPID)
(XEN)  - Virtual NMI
(XEN)  - MSR direct-access bitmap
(XEN)  - Unrestricted Guest
(XEN)  - VM Functions
(XEN)  - Virtualisation Exceptions
(XEN)  - Page Modification Logging
(XEN) HVM: ASIDs enabled.
(XEN) HVM: VMX enabled
(XEN) HVM: Hardware Assisted Paging (HAP) detected
(XEN) HVM: HAP page sizes: 4kB, 2MB, 1GB
(XEN) mwait-idle: max C-state count of 8 reached
(XEN) mwait-idle: max C-state count of 8 reached
(XEN) mwait-idle: max C-state count of 8 reached
(XEN) mwait-idle: max C-state count of 8 reached
(XEN) mwait-idle: max C-state count of 8 reached
(XEN) mwait-idle: max C-state count of 8 reached
(XEN) mwait-idle: max C-state count of 8 reached
```
And with the non-working kernel, it ends with : 
```
(XEN) HVM: ASIDs enabled.
(XEN) HVM: VMX enabled
(XEN) HVM: Hardware Assisted Paging (HAP) detected
(XEN) HVM: HAP page sizes: 4kB, 2MB, 1GB
(XEN) mwait-idle: max C-state count of 8 reached
(XEN) microcode: CPU2 updated from revision 0xc6 to 0xd6, date = 2020-04-23
(XEN) mwait-idle: max C-state count of 8 reached
(XEN) mwait-idle: max C-state count of 8 reached
(XEN) microcode: CPU4 updated from revision 0xc6 to 0xd6, date = 2020-04-23
(XEN) mwait-idle: max C-state count of 8 reached
(XEN) mwait-idle: max C-state count of 8 reached
```

**Solutions you've tried** *Temporary workaround*
- Live USB : Edit /boot/efi/EFI/qubes/xen.cfg to set the default kernel to the latest working
```
[global]
default=4.19.107-1.pvops.qubes.x86_64
```
- Reboot, downgrade microcode_ctl & reinstall the latest kernel : 
```
# qubes-dom0-update --action=downgrade microcode_ctl
# dracut --force --kver 4.19.125-1.pvops.qubes.x86_64
# cp /boot/initramfs-4.19.125-1.pvops.qubes.x86_64.img /boot/efi/EFI/qubes/
```
- Edit /boot/efi/EFI/qubes/xen.cfg to set the default kernel back to the latest
```
[global]
default=4.19.125-1.pvops.qubes.x86_64
```

**Comment by iamahuman** [comment]
You might want to do `sudo qubes-dom0-update --action=downgrade microcode_ctl`, although the chances of μcode update breaking the boot seem slim.

**Comment by iamahuman** [comment]
To properly bisect the problem, can you reinstall the latest (4.19.125-1) shipped kernel _with_ `microcode_ctl` downgraded? You can also invoke `dracut --kver 4.19.125-1.pvops.qubes.x86_64` without going through the hassle of downloading the entire kernel again. Not sure if `bin/kernel-install add 4.19.125-1.pvops.qubes.x86_64 /boot/vmlinuz-4.19.125-1.pvops.qubes.x86_64` does the job here though.

**Comment by marmarek** [comment]
Which exactly CPU is that? Skylake by any chance (Core gen 6)? I see there was yet another microcode update released just 3 days ago.

**Comment by gougeon-s** [comment]
That's a core 8th gen. I'll try to downgrade the microcode_ctl tomorrow

**Comment by zpc0** [comment]
FYI, some people reporting boot issues with 0xd6 microcode on the Core i5-8265U system.
https://launchpad.net/bugs/1880663
intel/Intel-Linux-Processor-Microcode-Data-Files#33

**Comment by gougeon-s** [comment]
I have reinstalled the latest kernel with microcode_ctl 2:2.1-29.qubes1.fc25 (downgraded) and it works.

**Comment by marmarek** [comment]
Thanks @zpc0 for finding the other reports. This indeed looks like the same issue.

**Comment by andrewdavidwong** [comment]
It looks like this is `not our bug`, but perhaps we want to keep this open for tracking purposes until the bug is fixed upstream.

**Comment by marmarek** [comment]
Just a little update: there is a [new microcode release from Intel](https://github.com/intel/Intel-Linux-Processor-Microcode-Data-Files/commit/0e4288f81f806620c65f70ee2bcf94b69d574096), but it doesn't include any update for Gen 8 CPUs.

**Comment by zpc0** [comment]
Intel has released new microcode. It contains update for Gen-8 mobile processor, so this issue may have been fixed.

**Comment by p1gp1g** [comment]
I still can't boot with *microcode_ctl-2.1-31* : 
```
(XEN) HVM: ASIDs enabled.
(XEN) HVM: VMX enabled
(XEN) HVM: Hardware Assisted Paging (HAP) detected
(XEN) HVM: HAP page sizes: 4kB, 2MB, 1GB
(XEN) mwait-idle: max C-state count of 8 reached
(XEN) microcode: CPU2 updated from revision 0xc6 to 0xde, date = 2020-05-18
(XEN) mwait-idle: max C-state count of 8 reached
(XEN) mwait-idle: max C-state count of 8 reached
(XEN) microcode: CPU4 updated from revision 0xc6 to 0xde, date = 2020-05-18
(XEN) mwait-idle: max C-state count of 8 reached
(XEN) mwait-idle: max C-state count of 8 reached
```

**Comment by iamahuman** [comment]
You can also try updating BIOS/firmware to see if it has shipped microcode updates, just in case it's Xen microcode loader's problem.

**Comment by p1gp1g** [comment]
Sorry for the late, I didn't want to do BIOS things during work period.
 
I have upgraded my BIOS to the latest version (1.9.1 for DELL latitude 5xxx) and microcode 0xde works. So it fixed since microcode v2.1-31.

Thanks
